packages/compiler/src/server/compile-service.ts

```diff
@@ -112,17 +119,22 @@ export function createCompileService({
    */
   async function compile(
     document: TextDocument | TextDocumentIdentifier,
+    additionalOptions?: CompilerOptions,
+    bypassCache: boolean = false,
```

Comment:
whats the purposse of bypassCache?

thoses options should also be an option bag not boolean positional

Reply:
When there is a cache, and it is the same as before, compiling will complete after parse the tsp files and emitting will be cancelled. 
So for emit-code request, we will bypassCache

Reply:
i mean why do you need to bypass cache

Reply:
Hello @timotheeguerin  when there is a program cache, and it is same as before, it will stop the whole compiling, the emitters will not be run, so the code is not emitted.

https://github.com/microsoft/typespec/blob/68dc81320736eb97a05cfd1b6d1f25f86598dffc/packages/compiler/src/core/program.ts#L265

and when CreateProgram return `shouldAbort:true`, the whole compiling will complete, the emitter is not run.
https://github.com/microsoft/typespec/blob/68dc81320736eb97a05cfd1b6d1f25f86598dffc/packages/compiler/src/core/program.ts#L173

But we need to emit code every time, so we need to bypass cache, in sure emitter is run everytime

Reply:
i see, can you just change the options to be an option bag then not positional booleans

Reply:
move those runtime options to option bag.